pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1114,10 +1114,13 @@ self: super: {
   # needed because of testing-feat >=0.4.0.2 && <1.1
   language-ecmascript = doJailbreak super.language-ecmascript;
 
-  # sexpr is old, broken and has no issue-tracker. Let's fix it the best we can. 
+  # sexpr is old, broken and has no issue-tracker. Let's fix it the best we can.
   sexpr =
     appendPatch (overrideCabal super.sexpr (drv: {
       isExecutable = false;
       libraryHaskellDepends = drv.libraryHaskellDepends ++ [self.QuickCheck];
     })) ./patches/sexpr-0.2.1.patch;
+
+  #
+  yi-core = super.yi-core.override { yi-language = self.yi-language_0_18_0; };
```

Comment:
The reason `yi-language` is at version `0.17.1` is because `configuration-hackage2nix.yaml` [specifies that](https://github.com/NixOS/nixpkgs/blob/e78bc3e38848d9a2b132612291869e4ed35fd626/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L2349). The reason it specifies that is because [`yi-core-0.17.1` is in the latest LTS](https://www.stackage.org/lts-12.5/package/yi-language). Since `yi-core` is apparantely [not in the LTS](https://www.stackage.org/lts-12.5/hoogle?q=yi-core) it defaults to the latest version which is 0.18.0. So your override is the right thing to do.

You could add a comment to instruct to remove the override once `yi-language >= 0.18` is in the LTS.